src/client/pmix_client.c

```diff
@@ -574,9 +585,21 @@ static void _putfn(int sd, short args, void *cbdata)
         /* shouldn't be possible */
         goto done;
     }
+#if defined(PMIX_ENABLE_DSTORE) && (PMIX_ENABLE_DSTORE == 1)
+    /* TODO: It is not safe to store data on a client side
+     * There is a possibility to get server/client conflict.
+     * Do nothing here misses PMIx_Get/PMIx_Put flow (w/o PMIx_Commit)
+     */
+    /*
+    if (PMIX_SUCCESS != (rc = pmix_dstore_store(ns->nspace, pmix_globals.myid.rank, kv))) {
+        PMIX_ERROR_LOG(rc);
+    }
+    */
+#else
```

Comment:
I agree with your statement about not storing data from the client side. However, why store this data in the dstore at all? It is solely for the local use of this client, so why not just store it in the hash table?

In other words, is it really necessary or advisable to treat the dstore and hash as mutually exclusive? If we allow "store_internal" to push data into the hash table (which we need to do as that data never should be circulated), then we need to provide a way to get it back. So if we retain both dstore and hash table support, we can avoid this problem.

Since you won't know where the data is located just on the basis of the "get" request, you will need to check both places. However, I wouldn't expect that to be too big of a penalty.